apps/css-workshop/pages/assets/theme.js

```diff
@@ -7,12 +7,12 @@ class ThemeButton extends HTMLElement {
     super();
     const html = /* html */ `
       <div class="settings-root">
-        <button title="Page settings" id="page-settings-menu-button" aria-label="Page settings" aria-haspopup="true" aria-controls="page-settings-menu">
+        <button tabindex="0" title="Page settings" id="page-settings-menu-button" aria-label="Page settings" aria-haspopup="true" aria-controls="page-settings-menu">
```

Comment:
This is interesting. Can you share your Safari version and macOS version?

Also, can you inspect the button to see that it does indeed have tabindex=0?

Lastly, can you check if you enabled "Enable Tab to highlight item on webpage" in Safari advanced settings?

Reply:
> Can you share your Safari version and macOS version?

I'm on Safari 16.6 and Mac Ventura 13.6.1.

> Also, can you inspect the button to see that it does indeed have tabindex=0?

Yes, it does have `tabindex=0`:

![image](https://github.com/iTwin/iTwinUI/assets/45748283/8df7620c-b5b1-44c4-a7fd-569f744f73d2)

> Lastly, can you check if you enabled "Enable Tab to highlight item on webpage" in Safari advanced settings?

I previously had "Keyboard Navigation" enabled in the Mac settings. With "Press Tab to highlight each item on a webpage" enabled in Safari along with or instead of "Keyboard Navigation" from the Mac, I received the same behavior.

Reply:
I wonder if this is specific to Safari <17. (I only tested it in 17).

Can you test our actual components too? e.g. click the button on https://itwin.github.io/iTwinUI/1844/react/?story=button--default&theme=dark and then log out `document.activeElement` in devtools. That's the part we care the most about.

/cc @FlyersPh9 Could you also test this change on your side?

Reply:
Upon confirming that, I realized that the focus is not changing upon clicking the button on my Safari version:


https://github.com/iTwin/iTwinUI/assets/45748283/35b33fc8-93eb-45f6-afb8-98f7ff10d6c4

Reply:
I just tested on this page: https://itwin.github.io/iTwinUI/1844/css/alert.html
The theme switcher button is opening for me!
- macOS Sonoma 14.4
- Safari 17.4

Reply:
Going to merge it, since this PR addresses the problem on the latest versions and is a net improvement over the previous situation (not working at all).

Fixing the problem on Safari 16 is too complicated and not worth investing time into imo.